{{cookiecutter.project_slug}}/server/{{cookiecutter.project_slug}}/core/views.py

```diff
@@ -66,29 +60,29 @@ class UserViewSet(
 
     @transaction.atomic
     def create(self, request, *args, **kwargs):
-        """
-        Endpoint to create/register a new user.
-        """
         serializer = UserRegistrationSerializer(data=request.data)
         serializer.is_valid(raise_exception=True)
-        serializer.save()  # This calls .create() on serializer
-        user = serializer.instance
-
+        user = serializer.save()
         # Log-in user and re-serialize response
         response_data = UserLoginSerializer.login(user, request)
         return Response(response_data, status=status.HTTP_201_CREATED)
 
     def update(self, request, *args, **kwargs):
-        """
-        Endpoint to create/register a new user.
-        """
         serializer = UserSerializer(data=request.data, instance=self.get_object(), partial=True)
-
         serializer.is_valid(raise_exception=True)
         serializer.save()
-        user = serializer.data
+        return Response(serializer.data, status=status.HTTP_200_OK)
 
-        return Response(user, status=status.HTTP_200_OK)
+    def destroy(self, request, *args, **kwargs):
+        """
+        When deleting a user's account, just disable their account first
+        The user may have a regret and try to get their account back
+        A background job should then properly delete the data after X days
+        """
+        user = self.get_object()
+        user.is_active = False
+        user.save()
+        return Response(status=status.HTTP_204_NO_CONTENT)
```

Comment:
Comment above should be clear.
By default apps shouldn't just delete important data from the DB like this.